contracts/DLCManager.sol

```diff
@@ -588,6 +598,21 @@ contract DLCManager is
         return _signerCount;
     }
 
+    function getFeeRatesForAddress(
+        address user
+    ) public view returns (uint256, uint256) {
+        uint256 mintFeeRate = btcMintFeeRates[user];
+        uint256 redeemFeeRate = btcRedeemFeeRates[user];
+
+        if (mintFeeRate == 0 && mintFeeRate != btcMintFeeRate) {
+            mintFeeRate = btcMintFeeRate;
+        }
+        if (redeemFeeRate == 0 && redeemFeeRate != btcRedeemFeeRate) {
+            redeemFeeRate = btcRedeemFeeRate;
+        }
```

Comment:
This check is like this because the default value of a mapping is 0. And 0 is a perfectly valid value to set a custom fee rate at. So we check _if it has been changed at all_.

Reply:
Makes sense that you want to handle the initial case of a 0, but i'm not sure this is the right way.

What if we roll this out, then we want to set a merchants fee rate to 0, wouldn't this still then force it to be the global value?